src/main/webapp/resources/product-details.js

```diff
@@ -0,0 +1,41 @@
+$(document).ready(function() {
+    console.log("loaded");
+
+    const template = Handlebars.compile($("#detailed-product-template").html());
+
+    const keyStorage = "my-cart";
+
+    function addToCart(product) {
+        let newCart = JSON.parse(localStorage.getItem(keyStorage)) || {};
+
+        let productCounter = newCart[product.productId];
+
+        if (productCounter) {
+            productCounter.count++;
+        } else {
+            productCounter = {
+                count: 1,
+                product: product
+            };
+        }
+        newCart[product.productId] = productCounter;
+        localStorage.setItem(keyStorage, JSON.stringify(newCart));
+    }
+
+    let productId = new URLSearchParams(location.search).get("product-id");
+
+    $.get(`/app/products/${productId}`, function(data, status, jqXHR) {
```

Comment:
The context part `/app` has to be extracted and defined globally since the context can be easily changed when redeploying.